src/backend/joanie/core/api.py

```diff
@@ -127,6 +137,124 @@ def perform_create(self, serializer):
         owner = models.User.objects.get_or_create(username=username)[0]
         serializer.save(owner=owner)
 
+    @transaction.atomic
+    def create(self, request, *args, **kwargs):
+        product_id = request.data.get("product")
+        course_code = request.data.get("course")
+        billing_address = request.data.get("billing_address")
+        credit_card_id = request.data.get("credit_card_id")
+
+        try:
+            product = Product.objects.get(uid=product_id)
+        except Product.DoesNotExist:
+            return Response(f"Product {product_id} does not exist.", status=404)
+
+        # If product is not free, we have to create a payment.
+        # To create one, a billing address is mandatory
+        if product.price.amount > 0 and billing_address is None:
```

Comment:
why not catch empty strings as well?
```suggestion
        if product.price.amount > 0 and not billing_address:
```